src/main/java/de/muenchen/isi/api/validation/EinrichtungstraegerValidator.java

```diff
@@ -22,11 +29,51 @@ public class EinrichtungstraegerValidator
      */
     @Override
     public boolean isValid(final InfrastruktureinrichtungDto value, final ConstraintValidatorContext context) {
-        return value != null && (einrichtungstraegerIsValid(value) || invalidEinrichtungstraegerAllowed(value));
+        return (value != null && einrichtungsTraegerValid(value));
     }
 
-    private boolean einrichtungstraegerIsValid(final InfrastruktureinrichtungDto value) {
-        return value.getEinrichtungstraeger() != null && value.getEinrichtungstraeger().getBezeichnung() != null;
+    private boolean einrichtungsTraegerValid(final InfrastruktureinrichtungDto value) {
+        if (!invalidEinrichtungstraegerAllowed(value)) {
+            if (value instanceof GrundschuleDto) {
+                var einrichtungstraeger = ((GrundschuleDto) value).getSchule().getEinrichtungstraeger();
+                return (
```

Comment:
Die beiden folgenden Logiken wiederholen sich mehrmals im Code und sind in eine eigene Methode auszulagern und Unit zu testen.

```java
 return (
                    this.isNotNull(einrichtungstraeger) &&
                    Einrichtungstraeger.getEinrichtungstraegerSchulen().contains(einrichtungstraeger)
                );

```

```java
                return (
                    this.isNotNull(einrichtungstraeger) &&
                    Einrichtungstraeger.getEinrichtungstraeger().contains(einrichtungstraeger)
                );

```